### PR TITLE
Move ipset acls into a different package.

### PR DESCRIPTION
**Pull request by 0xmchadha**
ipsets alcs creation, deletion, adding ips and deleting is moved into a different package.

### CODE REVIEW

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1376",
    "commit-sha": "38cc84bcbbf929e04c8d472bdd7a3bb7ca24158e"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "880",
    "commit-sha": "69412fa4fa1872ff487cec605fa56cccdd69d02b"
  }
]
```

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1376",
    "commit-sha": "38cc84bcbbf929e04c8d472bdd7a3bb7ca24158e"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "880",
    "commit-sha": "3a7f6c74d4b9aed53f2c9a8c59fb846192560116"
  }
]
```

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1376",
    "commit-sha": "38cc84bcbbf929e04c8d472bdd7a3bb7ca24158e"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "880",
    "commit-sha": "d191f77d318b3f6cca35eddd8bbf786366c58793"
  }
]
```

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1376",
    "commit-sha": "38cc84bcbbf929e04c8d472bdd7a3bb7ca24158e"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "880",
    "commit-sha": "e1fa106d19ce59a4f507b9de5473ebe8470aef7c"
  }
]
```

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1376",
    "commit-sha": ""
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "880",
    "commit-sha": "4299360f078b77250f0a05fbfa4883ccc668e906"
  }
]
```

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1376",
    "commit-sha": "709a0b764938e5c12dc0e517c13314223989dc9a"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "880",
    "commit-sha": "4299360f078b77250f0a05fbfa4883ccc668e906"
  }
]
```

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1376",
    "commit-sha": "709a0b764938e5c12dc0e517c13314223989dc9a"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "880",
    "commit-sha": "a64bb6d46dc164decf4f927f8e1c5b77616a8d97"
  }
]
```

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1376",
    "commit-sha": "709a0b764938e5c12dc0e517c13314223989dc9a"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "880",
    "commit-sha": "4db535e527091ea7b6a1ff12e3683cb077a47d21"
  }
]
```

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1376",
    "commit-sha": "df2813f4c674314133da8047a85bd59de447f9dc"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "880",
    "commit-sha": "4db535e527091ea7b6a1ff12e3683cb077a47d21"
  }
]
```

File: controller/pkg/ipsetmanager/ipsets.go [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+	ipsetPrefix           string
+	ipFilter              func(net.IP) bool
+	ipsetParams           *ipsetpackage.Params
+	toDestroy             []string
+}
+
+var lock sync.RWMutex
+var ipv4Handler *handler
+var ipv6Handler *handler
+
```
Comment (dstiliadis) [inline controller/pkg/ipsetmanager/ipsets.go:62]: Let's not use package wide variables. It always leads to a mess. Create an object and pack the variables in the object and initialize the object.

File: controller/internal/supervisor/supervisor.go [hunk 2]
```diff
@@ -207,6 +209,17 @@ func (s *Config) doCreatePU(contextID string, pu *policy.PUInfo) error {
...
+
+	iprules = append(iprules, pu.Policy.ApplicationACLs()...)
+	iprules = append(iprules, pu.Policy.NetworkACLs()...)
+
+	if err := ipsetmanager.RegisterExternalNets(contextID, iprules); err != nil {
+		s.Unlock()
+		zap.L().Error("Error creating ipsets for external networks", zap.Error(err))
+		return err
+	}
+
```
Comment (dstiliadis) [inline controller/internal/supervisor/supervisor.go:225]: Can't we do a RegisterPU method instead that takes the policy as argument and hides this inside it. No need for callers to know what to do for this.

File: controller/internal/supervisor/iptablesctrl/iptables.go [hunk 3]
```diff
@@ -305,10 +303,6 @@ func (i *iptables) DeleteRules(version int, contextID string, tcpPorts, udpPorts
 		zap.L().Warn("Failed to delete proxy sets", zap.Error(err))
 	}
 
-	// Destroy all the ACL related IPSets that were created
-	// on demand for any external services.
-	i.destroyACLIPsets(contextID)
-
 	return nil
```
Comment (dstiliadis) [inline controller/internal/supervisor/iptablesctrl/iptables.go:308]: Who removes them now? What is is this gone?

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1376",
    "commit-sha": "df2813f4c674314133da8047a85bd59de447f9dc"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "880",
    "commit-sha": "c31739d3591b6c93010a4e76e0e11c036a497809"
  }
]
```

File: controller/pkg/ipsetmanager/ipsets.go [hunk 4]
```diff
@@ -0,0 +1,397 @@
...
+const (
+	ipv4String = "v4-"
+	ipv6String = "v6-"
+)
+
+//GetManager is called to get the object to manage acls.
+func GetManager() ACLManager {
+	return &manager
+}
+
```
Comment (dstiliadis) [inline controller/pkg/ipsetmanager/ipsets.go]: That's the part I don't want. We still have a package global variable. There should be an object created at the top that is passed to both enforcer and supervisor. We haven't avoided the complications of the package wide shared variables with this.

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1422",
    "commit-sha": "cf67c0385dc4947fa9bfe78530535eb55eeb3f28"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "880",
    "commit-sha": "e3b73a6df15306e98056a5c41ff8fe22b526cc08"
  }
]
```

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1422",
    "commit-sha": "9849b0717e6421c9520f1ec4ed3c9e609ca0dfa3"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "880",
    "commit-sha": "d4773ed5de359dc78aa6f460db0a6f000663d162"
  }
]
```

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1422",
    "commit-sha": "97039fdbdfe1dd9972e955d730a4da2670ff98c3"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "880",
    "commit-sha": "e0b5f804950fa02db084544ac471651225acdced"
  }
]
```

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1422",
    "commit-sha": "d30a48c834b525830dddfb6b9119a9a2929fedd8"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "880",
    "commit-sha": "81d3692ad5dcc15353a0fb66bc8165e2327ba46a"
  }
]
```

**Comment by 0xmchadha** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1422",
    "commit-sha": "d30a48c834b525830dddfb6b9119a9a2929fedd8"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "880",
    "commit-sha": "1cdd9ca5dec1a035d101406df3866ebd632e2a78"
  }
]
```

**Review by dstiliadis — APPROVED** [review]

**Comment by 0xmchadha** [comment]
PR https://github.com/aporeto-inc/trireme-lib/pull/880 merged.

**Pull request merged**